pkg/cmd/operator/nodesetupdaemon.go

```diff
@@ -225,16 +241,27 @@ func (o *NodeConfigDaemonOptions) Run(streams genericclioptions.IOStreams, cmd *
 		return fmt.Errorf("can't create node config instance controller: %w", err)
 	}
 
-	// Start informers.
+	scyllaInformers.Start(ctx.Done())
 	namespacedKubeInformers.Start(ctx.Done())
-	nodeConfigInformers.Start(ctx.Done())
 	localNodeScyllaCoreInformers.Start(ctx.Done())
 	selfPodInformers.Start(ctx.Done())
 
-	// Run the controller to configure and reconcile pod specific options.
-	ncdc.Run(ctx)
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		nsc.Run(ctx)
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ntc.Run(ctx)
+	}()
 
-	<-ctx.Done()
+	wg.Wait()
```

Comment:
having this wait in defer is just a safety measure. It's here because this is the place where waitgroup should end before function explicitly returns. Leaving as is.